server/client.go

```diff
@@ -689,7 +689,12 @@ func (c *client) initClient() {
 	c.out.sg = sync.NewCond(&(c.mu))
 	opts := s.getOpts()
 	// Snapshots to avoid mutex access in fast paths.
-	c.out.wdl = opts.WriteDeadline
+	switch c.kind {
+	case ROUTER:
+		c.out.wdl = opts.Cluster.WriteDeadline
```

Comment:
Should this fall back to `opts.WriteDeadline` if the cluster variant is not set?

Reply:
The cluster variant has a default value `DEFAULT_FLUSH_DEADLINE` set by `setBaselineOptions`, so it doesn't much matter either way.

Reply:
Although I guess we should default to the global one for compatibility...